plugins/modules/virt_volume.py

```diff
@@ -0,0 +1,507 @@
+    xml: '{{ lookup("template", "pool/dir.xml.j2") }}'
+
+- name: Build a storage pool if it does not exist
+  community.libvirt.virt_pool:
+    command: build
+    name: vms
+
+- name: Start a storage pool
+  community.libvirt.virt_pool:
+    command: create
+    name: vms
+
+- name: List available pools
+  community.libvirt.virt_pool:
+    command: list_pools
+
+- name: Get XML data of a specified pool
+  community.libvirt.virt_pool:
+    command: get_xml
+    name: vms
+
+- name: Stop a storage pool
+  community.libvirt.virt_pool:
+    command: destroy
+    name: vms
+
+- name: Delete a storage pool (destroys contents)
+  community.libvirt.virt_pool:
+    command: delete
+    name: vms
+
+- name: Undefine a storage pool
+  community.libvirt.virt_pool:
+    command: undefine
+    name: vms
+
+- name: Gather facts about storage pools. Facts will be available as 'ansible_libvirt_pools'
+  community.libvirt.virt_pool:
+    command: facts
+
+- name: Gather information about pools managed by 'libvirt' remotely using uri
+  community.libvirt.virt_pool:
+    command: info
+    uri: '{{ item }}'
+  with_items: '{{ libvirt_uris }}'
+  register: storage_pools
+
+- name: Ensure that a pool is active (needs to be defined and built first)
+  community.libvirt.virt_pool:
+    state: active
+    name: vms
+
+- name: Ensure that a pool is inactive
+  community.libvirt.virt_pool:
+    state: inactive
+    name: vms
+
+- name: Ensure that a given pool will be started at boot
+  community.libvirt.virt_pool:
+    autostart: true
+    name: vms
+
+- name: Disable autostart for a given pool
+  community.libvirt.virt_pool:
+    autostart: false
+    name: vms
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+VIRT_FAILED = 1
+VIRT_SUCCESS = 0
+VIRT_UNAVAILABLE = 2
+
+ALL_COMMANDS = []
+ENTRY_COMMANDS = ['create', 'create_from', 'delete', 'download', 'get_xml', 'resize', 'upload',
+                  'wipe']
+HOST_COMMANDS = ['list_volumes', 'facts', 'info']
+ALL_COMMANDS.extend(ENTRY_COMMANDS)
+ALL_COMMANDS.extend(HOST_COMMANDS)
+
+ENTRY_STATE_ACTIVE_MAP = {
+    0: "inactive",
+    1: "active"
+}
+
+ENTRY_STATE_AUTOSTART_MAP = {
+    0: "no",
+    1: "yes"
+}
+
+ENTRY_STATE_PERSISTENT_MAP = {
+    0: "no",
+    1: "yes"
+}
+
+ENTRY_STATE_INFO_MAP = {
+    0: "inactive",
+    1: "building",
+    2: "running",
+    3: "degraded",
+    4: "inaccessible"
+}
+
+ENTRY_BUILD_FLAGS_MAP = {
+    "new": 0,
+    "repair": 1,
+    "resize": 2,
+    "no_overwrite": 4,
+    "overwrite": 8
+}
+
+ENTRY_DELETE_FLAGS_MAP = {
+    "normal": 0,
+    "zeroed": 1
+}
+
+ALL_MODES = []
+ALL_MODES.extend(ENTRY_BUILD_FLAGS_MAP.keys())
+ALL_MODES.extend(ENTRY_DELETE_FLAGS_MAP.keys())
+
+from ansible_collections.community.libvirt.plugins.module_utils.pool import LibvirtConnection as PoolConnection, HAS_VIRT, HAS_XML
+from ansible_collections.community.libvirt.plugins.module_utils.entry import EntryNotFound
+
+if HAS_VIRT:
+    import libvirt
+
+
+class LibvirtConnection(object):
```

Comment:
```suggestion
class LibvirtConnection(object):
```
don't you import this class from module_utils.pool?

Reply:
The  imported class is not called `LibvirtConnection` in the namespace of `plugin.modules.libvirt` which is something I adopted from https://github.com/flagbot/community.libvirt/tree/feature/virt_volume.